### DISCUSSION TITLE
Disable Notification-Controller

### DISCUSSION THREAD

**Discussion by ckotzbauer** [fluxcd/flux2#1714] [Q&A] [answered]
Hi,
is it possible to disable the notification-controller completely? My plan would be to get alerts of failing reconciliations via Prometheus rules and not through additional notifications. Is this something which is possible or is this a bad idea in general? 😄

Christian

**Comment by stefanprodan** [comment, marked as answer]
> Is it possible to disable the notification-controller completely?

Sure, remove it from `flux bootstrap --components` and all the other controllers will stop sending events to it. 

> My plan would be to get alerts of failing reconciliations via Prometheus rules and not through additional notifications.

Metrics will tell you that a reconciliation failed, not why it failed. While notification-controller will send you exact errors, Alertmanager will just say that something failed, then you have to look at the Flux objects status/events/logs to diagnose the issue.

**Reply by ckotzbauer** [reply]
> Sure, remove it from flux bootstrap --components and all the other controllers will stop sending events to it.

Great.

> Metrics will tell you that a reconciliation failed, not why it failed. While notification-controller will send you exact errors, Alertmanager will just say that something failed, then you have to look at the Flux objects status/events/logs to diagnose the issue.

Yes, that's correct. Maybe I will just try it if I'm fine with the limited info.

Thanks for your quick response. @stefanprodan
